app/views/handlers/list_action.py

```diff
@@ -47,7 +47,13 @@ def get_section_summary_url(self):
 
     def get_next_location_url(self):
         if self._return_to == "section-summary":
-            return self.get_section_summary_url()
+            if self.router.can_display_section_summary(
+                self.parent_location.section_id, self.parent_location.list_item_id
+            ):
+                return self.get_section_summary_url()
+
+            return self.router.get_section_resume_url(self._routing_path, resume=False)
```

Comment:
What scenario is this for?

Reply:
So this is interesting, firstly I couldn't see where this would happen in ccs, but I can in a couple of our test schemas, test_relationships being one. So if you want a test_schema to do an integration test you can use it with no changes. Add a person and go to the section summary and then add another and it will hit that separation condition you are looking for, because although section-summary is in the url, it naturally will stop the section summary appearing as the section is no longer complete.

Saying that it is edge and I don't think we do this outside of tests. Added to that I think the section endpoint should be handling this, it already goes to the right page its the resume panel thats unwanted, correct? This therefore feels to me we should be able to pass resume true/false to the section endpoint.

Reply:
I understand that, but current runner logic is to only return to the summary if it's possible immediately, else continue the normal journey. In the test_relationships example this would be to return to the parent list collector. There is a separate card in the backlog to look into only showing questions that need completing after navigating from a section summary.

Reply:
I've spoken to @MebinAbraham about this and it's been agreed with Donna that we can drop the logic about going to the next incomplete block if we can't go straight back to the summary. In that scenario it should go to the parent list collector (which it will if you remove the `get_section_resume_url` line).

Reply:
Ok, done, thanks👍